visualization/app/codeCharta/ui/dialog/dialog.addCustomConfigSettings.component.ts

```diff
@@ -44,22 +57,104 @@ export class DialogAddCustomConfigSettingsComponent implements FilesSelectionSub
+		this.downloadAndCollectPurgeableOldConfigs()
+
+		if (this.purgeableConfigs.size === 0) {
+			this.dialogService.showErrorDialog("Could not download and purge old configs automatically! Please try it by yourself.")
+			return
+		}
+
+		const confirmDialog = this.$mdDialog
+			.confirm()
+			.clickOutsideToClose(true)
+			.title("Confirm to purge old Configs")
+			.htmlContent("Are you sure to delete old Configs now?")
+			.ok("Purge Configs now!")
+			.cancel("Cancel")
+
+		try {
+			await this.$mdDialog.show(confirmDialog)
+			// The user has confirmed to purge old Configs.
+			this.purgeOldConfigs()
+		} catch {
+			// The user has cancelled the purge - do nothing.
+		}
+	}
+
+	purgeOldConfigs() {
+		if (!this.purgeableConfigs.size) {
+			return
+		}
+
+		CustomConfigHelper.deleteCustomConfigs([...this.purgeableConfigs])
+	}
+
+	downloadAndCollectPurgeableOldConfigs() {
+		this.purgeableConfigs.clear()
+		const customConfigs = CustomConfigHelper.getCustomConfigs()
+
+		const downloadableConfigs: Map<string, ExportCustomConfig> = new Map()
+		const daysPerMonth = 30
+
+		for (const [key, value] of customConfigs.entries()) {
+			if (value?.creationTime === undefined) {
+				// Fallback, if creationTime property is not present. This can happen because it was released later.
+				value.creationTime = Date.now()
+			}
+
+			// Download e.g. 6 month old or older Configs.
+			const ageInMonth = (Date.now() - value.creationTime) / (1000 * 60 * 60 * 24 * daysPerMonth)
+			if (ageInMonth < this.customConfigAgeLimitInMonths) {
+				continue
+			}
+
+			downloadableConfigs.set(key, CustomConfigHelper.createExportCustomConfigFromConfig(value))
+			this.purgeableConfigs.add(value)
+		}
+
+		if (downloadableConfigs.size > 0) {
+			CustomConfigHelper.downloadCustomConfigs(downloadableConfigs, this.customConfigFileStateConnector)
+		}
+	}
+
 	validateCustomConfigName() {
 		if (
-			CustomConfigHelper.hasCustomConfig(
+			CustomConfigHelper.hasCustomConfigByName(
 				this.customConfigFileStateConnector.getMapSelectionMode(),
 				this.customConfigFileStateConnector.getSelectedMaps(),
 				this._viewModel.customConfigName
 			)
 		) {
-			this._viewModel.addWarningMessage = '<i class="fa fa-warning"></i> A Custom Config with this name already exists.'
+			this._viewModel.addErrorMessage = "A Custom Config with this name already exists."
+		} else {
+			this._viewModel.addErrorMessage = ""
+		}
+	}
+
+	validateLocalStorageSize() {
+		let allStringsConcatenated = ""
+		for (const [key, value] of Object.entries(localStorage)) {
+			allStringsConcatenated += key + value
+		}
+
+		// It does not exist a limit for the total localStorage size that applies to all browsers.
+		// Usually 2MB - 10MB are available (5MB seems to be very common).
+		// The localStorage size (e.g. 5MB) is assigned per origin.
+		// Multiply localStorage characters by 16 (bits( because they are stored in UTF-16.
+		// Add 3KB as it seems there is some default overhead.
```

Comment:
We concatenated the key and values of `localStorage` (values of local storage are [DOMStrings](https://developer.mozilla.org/en-US/docs/Web/API/DOMString), meaning 16-bit unsigned integers interpreted as UTF-16 code in JS). I guess the "+3" is the overhead of the localStorage object itself :)